src/inapp/utils.ts

```diff
@@ -114,6 +115,103 @@ export const sortInAppMessages = (messages: Partial<InAppMessage>[] = []) => {
   return messages.sort(by(['priorityLevel', 'asc'], ['createdAt', 'asc']));
 };
 
+/**
+ * detect amount of local storage remaining (quota) and used (usage).
+ * if usageDetails exist (not supported in Safari), use this instead of usage.
+ */
+export const determineRemainingStorageQuota = async () => {
+  try {
+    const storage: BrowserStorageEstimate | undefined =
+      'storage' in navigator && 'estimate' in navigator.storage
+        ? await navigator.storage.estimate()
+        : undefined;
+
+    /** 50 MB is the typical web browser cache quota for mobile devices */
+    const mobileBrowserQuota = 52428800;
+    /** max quota of browser storage that in-apps will potentially fill */
+    const inAppMaxBrowserQuota = storage?.quota && storage.quota * 0.6;
```

Comment:
Is the 0.6 of the quota documented somewhere as the max we should use or a little arbitrary with the initial release? Could we put some comments on how we got that number?